cms/private_media/views.py

```diff
@@ -1,34 +1,252 @@
+from collections.abc import Sequence
 from typing import TYPE_CHECKING
 
-from django.core.exceptions import PermissionDenied
-from django.shortcuts import redirect
-from wagtail.images.permissions import permission_policy
-from wagtail.images.views.serve import ServeView
+from django.conf import settings
+from django.core.exceptions import PermissionDenied, SuspiciousOperation
+from django.http import FileResponse, Http404, HttpResponse
+from django.shortcuts import get_object_or_404, redirect
+from django.utils.cache import add_never_cache_headers, patch_cache_control
+from django.views.generic import View
+from wagtail.documents import get_document_model
+from wagtail.documents.models import document_served
+from wagtail.documents.permissions import permission_policy as document_permission_policy
+from wagtail.images import get_image_model
+from wagtail.images.exceptions import InvalidFilterSpecError
+from wagtail.images.models import SourceImageIOError
+from wagtail.images.permissions import permission_policy as image_permission_policy
+from wagtail.images.utils import verify_signature
 
 if TYPE_CHECKING:
-    from django.http import HttpResponse
+    from django.http import HttpRequest, HttpResponseBase, HttpResponseRedirect
+    from wagtail.documents.models import AbstractDocument
+    from wagtail.images.models import AbstractImage, AbstractRendition
 
-    from cms.private_media.models import AbstractPrivateRendition, PrivateImageMixin
 
+class ImageServeView(View):
```

Comment:
There's a lot of new code here copied from Wagtail, which is going to be a burden to maintain as packages upgrade. 

What are the differences between this and `ServeView`? It looks like the old view could handle changing how to serve public / private images by changing functionality in `serve`?

Reply:
As outlined in the description, we want to use different cache headers etc in different scenarios, but `ServeView` adds it's 'public' ones for everything, which isn't really suitable for this project.

Reply:
We're also doing the permission check / rejection here before attempting to find/create the rendition, which I think is slightly nicer.

Reply:
Oh, and the Wagtail version also raises a `PermissionDenied` exception when the signature is invalid - which is not the best choice, and could definitely make things difficult to reason about should we face any issues in an environment. Raising `SuspiciousOperation` instead helps to resolve that.